rfcs/NNNN-hot-elements.md

```diff
@@ -0,0 +1,77 @@
+---
+Status: Active
+Champions: {{ rictic }}
+PR: {{ 15 }}
+---
+
+# Hot Elements
+
+A new Lit Labs package for supporting hot module replacement with custom elements.
+
+## Objective
+
+### Goals
+- Instantiating a new custom element, post HMR, should use the newest version of the constructor.
```

Comment:
Can we have some user-focused goals? ie, editing a template or CSS should result in a UI update in XXms, with stable DOM state, etc.

Reply:
Wrote it up as:

> In a LitElement that accepts HMRs, editing a template or style and saving should result in the page updating by doing the minimum amount of work needed.

Reply:
Nit: Worth explicitly calling out the expectations around state preservation during HMR?

Reply:
Good point, done!

Reply:
Good point, done!

Reply:
I feel like this list could benefit from a TL;DR type bullet at the beginning, ala "Add an API that allows a compatible dev server to detect a file change on disk and notify a module containing a LitElement definition to re-render any existing instances" (only using the correct words :P). Otherwise looks great!